src/rules/objectLiteralShorthandRule.ts

```diff
@@ -63,6 +98,32 @@ function walk(ctx: Lint.WalkContext<void>) {
     });
 }
 
+function fixShorthandToLonghand(node: ts.ShorthandPropertyAssignment | ts.MethodDeclaration): Lint.Fix {
+    const isGenerator = (node as ts.MethodDeclaration).asteriskToken !== undefined;
+    const isAsync = hasModifier(node.modifiers, ts.SyntaxKind.AsyncKeyword);
+    let
+    replacementStart = (isAsync && node.modifiers !== undefined) ? getModifier(node, ts.SyntaxKind.AsyncKeyword)!.getStart() : -1;
+    replacementStart = (isGenerator && !isAsync) ? (node as ts.MethodDeclaration).asteriskToken!.getStart() : -1;
+    replacementStart = replacementStart === -1 ? node.name.getStart() : replacementStart;
+
+    const fixes: Lint.Fix = [
+        isMethodDeclaration(node)
+            ? Lint.Replacement.replaceFromTo(
+                    replacementStart,
+                    node.name.end,
+                    `${node.name.getText()}:${isAsync ? " async" : ""} function${isGenerator ? "*" : ""}`,
+                )
+            /* tslint:disable:trailing-comma */
+            : Lint.Replacement.appendText(node.name.getStart(), `${node.name.text}: `)
```

Comment:
consider moving the fix for ShorthandPropertyAssignment out of the function. It will make many conditions and type assertions unnecessary